### PR TITLE
Remove ResourceLimitsPriorityFunction feature gate in the scheduler

### LINKED ISSUE #91817: Deprecate ResourceLimitsPriorityFunction feature in the scheduler
[ResourceLimitsPriorityFunction](https://github.com/kubernetes/kubernetes/blob/master/pkg/features/kube_features.go#L200:2) has bee in alpha since 1.9. Since then, no requests were made to graduate the feature, which indicates lack of usage. Based on that I propose we deprecate this scoring function.

/cc @aveshagarwal
/sig scheduling

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**
/kind deprecation

**What this PR does / why we need it**:
`ResourceLimitsPriorityFunction` has bee in alpha since 1.9.  Since then, no requests were made to graduate the feature, which indicates lack of usage. Based on that we prefer to remove this scoring function.

**Which issue(s) this PR fixes**:

Fixes #91817

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Scheduler's alpha feature 'ResourceLimitsPriorityFunction' is completely removed due to lack of usage
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -195,8 +195,11 @@ const (
 
 	// owner: @aveshagarwal
 	// alpha: v1.9
+	// deprecated: v1.19
 	//
 	// Enable resource limits priority function
+	//
+	// This feature is deprecated, and will be removed in a future release.
```
Comment (ahg-g) [inline pkg/features/kube_features.go:202]: @liggitt we can remove it in this release since it is alpha, right?

Comment (liggitt) [reply pkg/features/kube_features.go:202]: yes, it can be removed directly. make sure the release note describes the removal and the current alpha state.

**Comment by ahg-g** [comment]
pls update the release note as follows: "Scheduler's alpha feature 'ResourceLimitsPriorityFunction' is completely removed due to lack of usage"

/approve

/assign @liggitt

**Pull request merged**